core/controllers/domain_objects_validator.py

```diff
@@ -86,32 +106,78 @@ def validate_change_dict_for_blog_post(change_dict):
             raise Exception(
                 'Invalid tags provided. Tags not in default tags list.')
 
+    return change_dict
 
-def validate_collection_change(obj):
+
+def validate_collection_change(change_dict):
     # type: (Dict[String, Any]) -> None
     """Validates collection change.
 
     Args:
-        obj: dict. Data that needs to be validated.
+        change_dict: dict. Data that needs to be validated.
+
+    Returns:
+        CollectionChange. Returns CollectionChange object after validation.
     """
     # No explicit call to validate_dict method is necessary, because
     # CollectionChange calls validate method while initialization.
-    collection_domain.CollectionChange(obj)
+    return collection_domain.CollectionChange(change_dict)
 
 
-def validate_email_dashboard_data(data):
+def validate_email_dashboard_data(email_dashboard_data_dict):
     # type: (Dict[String, Optional[Union[bool, int]]]) -> None
-    """Validates email dashboard data.
+    """Validates the email dashboard data dict.
 
     Args:
-        data: dict. Data that needs to be validated.
+        email_dashboard_data_dict: dict. Data that needs to be validated.
+
+    Returns:
+        dict(str, any). Returns email dashboard data dict after validation.
     """
+    # The method returns a dict containing fields of email dashboard
+    # query params. This dict represents the UserQueryParams class, which is a
+    # namedtuple. Hence the fields are validating as a part of schema validation
+    # before saving new user queries in the handler.
     predicates = constants.EMAIL_DASHBOARD_PREDICATE_DEFINITION
     possible_keys = [predicate['backend_attr'] for predicate in predicates]
 
-    for key, value in data.items():
+    for key, value in email_dashboard_data_dict.items():
         if value is None:
             continue
         if key not in possible_keys:
             # Raise exception if key is not one of the allowed keys.
             raise Exception('400 Invalid input for query.')
+
+    return email_dashboard_data_dict
+
+
+def validate_task_entry_dict(task_entry_dict):
+    # type: (Dict[String, Any]) -> None
+    """Validates the task entry dict.
+
+    Args:
+        task_entry_dict: dict. Data that needs to be validated.
+
+    Returns:
+        dict(str, any). Returns task_entry_dict after validation.
+    """
+    # For creating the TaskEntry domain object, we have to include the
+    # exploration_id and the user_id which are not included in the
+    # task_entry_dict thus it might not be possible to create the domain object.
```

Comment:
...in the task_entry_dict. Thus, it is not possible to create the full domain object at the payload validation stage. Hence, the key-value pairs of task_entry_dict are being validated as a part of schema validation.

Reply:
Thank you for review, will take care of these points while creating another PR.